packages/glider/src/index.js

```diff
@@ -0,0 +1,130 @@
+ *
+ * Glider-JS doesn't support RTL at the moment, this is to add basic
+ * functioning support for the nav arrows as otherwise the nav arrows
+ * becomes useless on RTL sites.
+ *
+ * @todo Maybe replace glider-js with other lightweight lib which has RTL support. or Replace it with 'amp-carousel' once we have the support.
+ *
+ * @param {Object|string} slide Slide arrow string based on action.
+ * @param {boolean}       dot   Is dot navigation action.
+ * @param {Object}        e     Event object.
+ *
+ * @return {boolean} Navigation done.
+ */
+Glider.prototype.scrollItem = function (slide, dot, e) {
+  // glider-js doesn't seem to pass right amount of arguments.
+  if (e === undefined && dot?.target) {
+    e = dot;
+    dot = false;
+  }
+
+  if (e === undefined) {
+    // Somehow we ended up triggering this function twice. Abort to prevent scrolling back and forth.
+    return false;
+  }
+
+  if (e) {
+    e.preventDefault();
+  }
+
+  // Somehow slidesToScroll and slidesToShow can end up being 0.
+  this.opt.slidesToScroll = Math.max(1, this.opt.slidesToScroll);
+  this.opt.slidesToShow = Math.max(1, this.opt.slidesToShow);
+  // This will also cause this.itemWidth to be Infinity because division by zero returns Infinity in JS.
+  // Update this.itemWidth with actual value in this case.
+  if (this.itemWidth === Infinity) {
+    // It's a sibling.
+    const carouselWrapper = e.target.parentElement.querySelector(
+      '.web-stories-list__carousel'
+    );
+    const itemStyle = window.getComputedStyle(
+      carouselWrapper.querySelector('.web-stories-list__story')
+    );
+
+    this.itemWidth =
+      parseFloat(itemStyle.width) +
+      (parseFloat(itemStyle.marginLeft) + parseFloat(itemStyle.marginRight));
+  }
+
+  const originalSlide = slide;
+  ++this.animate_id;
+
+  if (dot === true) {
+    slide = slide * this.containerWidth;
+    slide = Math.round(slide / this.itemWidth) * this.itemWidth;
+  } else {
+    if (typeof slide === 'string') {
+      const backwards = slide === 'prev';
+
+      // use precise location if fractional slides are on
+      if (this.opt.slidesToScroll % 1 || this.opt.slidesToShow % 1) {
+        slide = this.getCurrentSlide();
+      } else {
+        slide = !isNaN(this.slide) ? this.slide : 0;
+      }
+
+      if (backwards) {
+        slide -= this.opt.slidesToScroll;
+      } else {
+        slide += this.opt.slidesToScroll;
+      }
+
+      if (this.opt.rewind) {
+        const scrollLeft = this.ele.scrollLeft;
+        slide =
+          backwards && !scrollLeft
+            ? this.slides.length
+            : !backwards &&
+              scrollLeft + this.containerWidth >= Math.floor(this.trackWidth)
+            ? 0
+            : slide;
+      }
+    }
+
+    slide = Math.min(slide, this.slides.length);
+
+    this.slide = slide;
+    slide = this.itemWidth * slide;
+  }
+
+  this.scrollTo(
+    slide,
+    this.opt.duration * Math.abs(this.ele.scrollLeft - slide),
+    function () {
```

Comment:
```suggestion
   () => {
```

Reply:
Don't think it will work due to usage of `this`.

Also, this is just existing code moved to a different place, not new code.

Reply:
Okay never mind.